csrc/ir/utils.cpp

```diff
@@ -892,39 +892,70 @@ void validateDomainEquivalence(
       std::any_of(frontier.begin(), frontier.end(), is_symb);
   bool dom1_has_symbolic =
       std::any_of(dom1_set.begin(), dom1_set.end(), is_symb);
+
+  CompareDomainResult result;
+
+  auto check_ids = [](const auto& ids_to_check,
+                      const auto& target_set) -> bool {
+    bool unreachable = false;
+    for (auto id : ids_to_check) {
+      // Symbolic and broadcast IDs are ignored
+      if (id->template as<IterDomain>()->getIterType() == IterType::Symbolic ||
+          id->template as<IterDomain>()->isBroadcast()) {
+        continue;
+      }
+      if (!target_set.count(id)) {
+        // not found in target, which means either:
+        //
+        // 1. id is unreachable from target_set, or
+        // 2. id is reachable from target_set but was erased from
+        // target_set as it was used as an input in the traversal.
+        //
+        // The second case means id is redudant
```

Comment:
Is this added primarily to check that the loop domain could have IDs not covered by the logical domain, but these additional IDs must be disconnected, instead of being a duplicated ID.

Reply:
Right. I was wondering if that check would need to be done by this function itself since it seems like it's more about the validity of a loop domain, which should not include any duplicated IDs, but detection of duplication is also done by the original function, I decided to keep it detectable.